mypy/plugin.py

```diff
@@ -252,6 +252,13 @@ def named_type(self, qualified_name: str, args: Optional[List[Type]] = None) ->
         """Construct an instance of a builtin type with given type arguments."""
         raise NotImplementedError
 
+    @abstractmethod
+    def named_type_or_none(self,
+                           qualified_name: str,
+                           args: Optional[List[Type]] = None) -> Optional[Instance]:
+        """Construct an instance of a builtin type with given type arguments."""
```

Comment:
Docstring is the name as `named_type`, can you say what's different about this method?

Reply:
I'm afraid I myself do not know what the difference between these methods is, @ilevkivskyi advised (https://github.com/python/mypy/pull/8578#discussion_r399578066) to use `named_type_or_none` and I had to add it to `SemanticAnalyzerPluginInterface`.

Reply:
suggest docstring:

```
"""Construct an instance of a type with given type arguments, or None if a type could not be constructed for the qualified type name. This is possible when the qualified name includes a module name and the module has not been imported."""
```
From looking through the code it seems `named_type_or_none` is useful when `qualified_name` is NOT a built-in type. Things like `typing._TypedDict`, `abc.ABCMeta`, etc. whereas `named_type` is almost exclusively used for `__builtins__.*` or `builtins.*` (the one exception is `m.Signal`). So my assumption is that `named_type_or_none` is useful when the qualified type name is includes a module name which may not be known. This matches with the reason why `named_type_or_none` was added to `SemanticAnalyzerPluginInterface` in this PR in the first place - exception when importing as `from attr import attrs, Attribute` but not when `import attrs` and referring to `attrs.Attribute`.

However since I am not an expert on mypy internals, and the `named_type_or_none` method does not have a docstring in `SemanticAnalyzerInterface` I'm afraid I'm rather underqualified to make the docstring and can't be certain about my interpretations.